core/templates/components/entity-creation-services/story-creation.service.spec.ts

```diff
@@ -95,19 +99,19 @@ describe('Story Creation Service', () => {
         urlFragment: 'url'
       })
     });
+    spyOn(mockWindow, 'open');
+    spyOn(mockWindow.location, 'reload');
 
     $httpBackend.expectPOST('/topic_editor_story_handler/id')
       .respond(200, {storyId: 'id'});
 
-    expect(mockWindow.location).toBe('');
-
-
     StoryCreationService.createNewCanonicalStory();
     $scope.$apply();
 
     $httpBackend.flush();
 
-    expect(mockWindow.location).toBe('/story_editor/id');
+    expect(mockWindow.open).toHaveBeenCalledWith('/story_editor/id');
```

Comment:
The previous test was stronger (it's usually better to have a 'toBe' or 'toEqual' check than 'toHaveBeenCalledWith' which is weaker). Why change it?

Reply:
I have changed the behaviour a little bit as follows:

Previously, after creating a new story, the page shifted to the story editor page and the other opened tabs of the same url could not detect staleness because for detection of staleness, the functionalities are present in topic editor pae component. So, now, on creation of a new story, the story editor gets opened in a new tab and the topic editor page gets opened again in the existing tab.

Reply:
That's why I have changed the tests here.

Reply:
I'm sorry, I don't understand. It seems really odd to change the user flow completely because of the implementation of a feature like this.

From the user's perspective, can you describe the issue that means that the new flow is better than the old? Always start from the user's perspective, not the technical perspective.

Reply:
I have reverted back to the old user flow now and used a new service to show and hide the modals.